src/app/(space)/(content)/[[...pathname]]/page.tsx

```diff
@@ -114,24 +124,38 @@ export async function generateViewport({ params }: { params: PagePathParams }):
     };
 }
 
-export async function generateMetadata({ params }: { params: PagePathParams }): Promise<Metadata> {
+export async function generateMetadata({
+    params,
+    searchParams,
+}: {
+    params: PagePathParams;
+    searchParams: { fallback?: string };
+}): Promise<Metadata> {
     const { space, pages, page, customization, parent } = await fetchPageData(params);
-    if (!page) {
+    const canFallback = searchParams.fallback?.toLocaleLowerCase() === 'true';
+
+    let targetPage = page;
+    if (!page && canFallback) {
+        const rootPage = resolveFirstDocument(pages, []);
+        targetPage = rootPage?.page;
+    }
+
+    if (!targetPage) {
```

Comment:
Yes good point. Reverting this change.

Reply:
Actually this is needed otherwise during the first `fallback=true` request to the non-existing page, `generateMetadata` will short-circuit and return a Page not found here:
https://github.com/GitbookIO/gitbook/blob/81b372f372e52d59ad944451c03f7d33f8c303c1/src/app/(space)/(content)/%5B%5B...pathname%5D%5D/page.tsx#L119

And we still need to generate the metadata for that request I think hence why I use the fallback page for it. Yes they will be generated again anyway in the next request (redirect) but I think we still need to generate the metadata for that first request.